pypsi/commands/include.py

```diff
@@ -65,28 +67,53 @@ def run(self, shell, args):
 
     def include_file(self, shell, path):
         fp = None
-        ifile = IncludeFile(path)
+        self.ifile = IncludeFile(path)
 
         if self.stack:
             for i in self.stack:
-                if i.abspath == ifile.abspath:
+                if i.abspath == self.ifile.abspath:
                     self.error(shell, "recursive include for file ",
-                               ifile.abspath, '\n')
+                               self.ifile.abspath, '\n')
                     return -1
 
-        self.stack.append(ifile)
+        self.stack.append(self.ifile)
 
         try:
             fp = safe_open(path, 'r')
         except (OSError, IOError) as e:
             self.error(shell, "error opening file ", path, ": ", str(e), '\n')
+            self.stack.pop()
             return -1
 
-        for line in fp:
-            shell.execute(line.strip())
-            ifile.line += 1
+        try:
+            self.lines = [line for line in fp]
```

Comment:
This is load the entire file into memory, which isn't ideal. It'd be nice to read the file line by line.